server/src/main/scala/cromwell/CromwellEntryPoint.scala

```diff
@@ -78,13 +79,26 @@ object CromwellEntryPoint extends GracefulStopSupport {
     val cromwellClient = new CromwellClient(args.host, "v2")
 
     val singleSubmission = validateSubmitArguments(args)
-    val submissionFuture = () => cromwellClient.submit(singleSubmission).andThen({
-      case Success(submitted) =>
-        Log.info(s"Workflow ${submitted.id} submitted to ${args.host}")
-      case Failure(t) =>
-        Log.error(s"Failed to submit workflow to cromwell server at ${args.host}")
-        Log.error(t.getMessage)
-    })
+    val submissionFuture = () => {
+      val io = cromwellClient.submit(singleSubmission).value.attempt flatMap {
+        case Right(Right(submitted)) =>
```

Comment:
Is this is like this because we are examining the result of a two-part operation which could fail at either step?

Reply:
Yes. The server's `HttpResponse` could come back with an error, or we could have an error even connecting for some reason creating an `Exception`.